containers/go-postgres/.devcontainer/.env

```diff
@@ -0,0 +1,3 @@
+POSTGRES_USER=postgres
```

Comment:
@juzuluag - I wonder if this is how that information is passed in the hello world go test at the bottom of this PR?

Reply:
@gfmatthews  is correct. If you look at Lines 21-25 in the `docker-compose.yml` you'll see that these are getting passed to the environment for the app. They also get passed to the database. We want these to be the same, so having a .env file seemed easiest.

Reply:
What I've changed is to have the docker-compose use the .env directly instead of individual variables. Feels like a neater way to ensure that the same variables are being used everywhere.

Reply:
@gfmatthews , the fact there is a `.env` at the same location of the `docker-compose.yml` it allowed to read those when `docker-compose up` is invoked. So at the container level, those environment variable would exist, then the golang app (`hello.go`) was able to find them. 
But with these new changes from @KanishkT123, it is cleaner imho.